src/intrprtr.c

```diff
@@ -1975,24 +1935,20 @@ void            IntrIntExpr (
 **  'IntrLongIntExpr' is the action to  interpret a long literal integer
 **  expression whose digits are stored in a string GAP object.
 */
-void            IntrLongIntExpr (
-    Obj               string )
+void IntrLongIntExpr(Obj string)
 {
-    Obj                 ret;            /* integer encoded as GAP obj      */
-    /* ignore or code                                                      */
     if ( STATE(IntrReturning) > 0 ) { return; }
     if ( STATE(IntrIgnoring)  > 0 ) { return; }
-    if ( STATE(IntrCoding)    > 0 ) { CodeLongIntExpr( string ); return; }
 
-    ret = IntStringInternal(string);
-    
-    if ( ret == Fail ) {
-        /* This should never happen */
-        ErrorQuit("Int: Invalid parsing (internal error)", 0, 0);
```

Comment:
I am a little nervous about dumping this entirely -- while I agree it should never happen, it could also cause a serious byte-code problem if it ever did.

Perhaps a `GAP_ASSERT(ret != Fail)`?

Reply:
For coding, such an asser already exists in `CodeIntExpr`. For interpretation, I don’t see a problem.

Finally, the removed Error check itself is fairly recent, before that (and in the other code paths), we never made comparable checks. And really, if somebody breaks the parser badly enough to let non-digits reach this point, we’d be in deep trouble and notice anyway.

Reply:
Specifically, you added that “paranoia” check two years ago, see https://github.com/gap-system/gap/commit/d4e4d740c4a06f92ec8a05e1fa463a02977fbfbe

Reply:
That would explain why I liked it :) if you are still really it can't occur, then I am happy to leave it out.

Reply:
Yes, I am sure it can't happen at that point. And even if I am totally wrong, then this will be immediately visible and easy to debug, as `fail` would occur in a place where an integer object is expected (and that would have been triggered by a bug in the GAP parser).

That said, being extra paranoid and adding the two `GAP_ASSERT` doesn't hurt, so I just did that.